### DISCUSSION TITLE
Storage Plugin: `migrate()` - When/Where to use?

### DISCUSSION THREAD

**Discussion by allidoisace** [ionic-team/capacitor#5129] [Q&A] [answered]
> Storage
> Data migration required! The internal storage mechanism has changed and requires data migration. A convenience method has been added: migrate(). To update your app without affecting end users, call migrate() before any other methods.

I am unsure what it means by "...call migrate() before any other methods." Does this mean that `migrate()` should be called every time the app starts up?

**Comment by jcesarmobile** [comment, marked as answer]
Should be called before you want to read values from storage. It really only needs to run once, but nothing happens if you run it every make code simpler, if not you’ll have to store some value to know that you already migrated. 
Also, there is a function to delete the migrated values, because if you don’t delete them and delete the migrated value it will re appear.
The migration is only needed for iOS and web, on android nothing changed.
